File: app/src/main/java/com/example/hzg/mysussr/widget/LoadingDialog.java
package com.example.hzg.mysussr.widget;

import android.view.Gravity;

import com.example.hzg.mysussr.R;
import com.example.hzg.mysussr.base.BaseDialog;
import com.example.hzg.mysussr.databinding.DialogLoadingBinding;

/**
 * Created by hzg on 2018/4/6.
 */

public class LoadingDialog extends BaseDialog {
    private DialogLoadingBinding mBinding;
    private String message;

    public static LoadingDialog newInstance(String message, Boolean cancelable) {
        LoadingDialog dialog = new LoadingDialog();
        dialog.setMessage(message);
        dialog.setCancelable(cancelable);
        return dialog;
    }

    @Override
    public int getLayoutId() {
        return R.layout.dialog_loading;
    }

//    @Override
//    protected int getLayoutResId() {
//        return R.layout.dialog_loading;
//    }

    @Override
    protected void initDialog() {
      setGravity(Gravity.CENTER);
        setFullScreen(true);
    }

    @Override
    protected void init() {
        mBinding = getDataBinding();
        mBinding.tvContent.setText(message);
    }

    public void setMessage(String message) {
        this.message = message;
    }


}

File: app/src/main/java/com/example/hzg/mysussr/widget/LoadingDialog.kt
package com.example.hzg.mysussr.widget

import android.view.Gravity

import com.example.hzg.mysussr.R
import com.example.hzg.mysussr.base.BaseDialog
import com.example.hzg.mysussr.databinding.DialogLoadingBinding

/**
 * Created by hzg on 2018/4/6.
 *
 */

class LoadingDialog : BaseDialog() {

    private var message: String? = null

    override fun getLayoutId(): Int {
        return R.layout.dialog_loading
    }

    override fun initDialog() {
        setGravity(Gravity.CENTER)
        setFullScreen(true)
    }

    override fun init() {
        with(getDataBinding<DialogLoadingBinding>()) {
            tvContent.text = message
        }

    }

    fun setMessage(message: String) {
        this.message = message
    }

    companion object {

        fun newInstance(message: String, cancelable: Boolean?): LoadingDialog {
            val dialog = LoadingDialog()
            dialog.setMessage(message)
            dialog.isCancelable = cancelable!!
            return dialog
        }
    }


}
